cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -667,96 +667,164 @@ fn parse_url(
 mod tests {
     use super::*;
 
-    #[test]
-    fn registry_try_parse_valid() {
-        let tests = vec![
-            (
-                "http://localhost:8080",
-                Registry {
+    struct TestCase {
+        input: String,
+        expected_registry: Registry,
+        expected_registry_host_port_string: String,
+        expected_registry_host_port_namespaced_string: String,
+    }
+
+    fn setup_test_cases() -> Vec<TestCase> {
```

Comment:
I think this should have `#[cfg(test)]` so it only compiles when testing

Reply:
also feels like a `make_test_cases` or `get_test_cases` rather than `setup_` which I'd expect to setup _something_ for the tests, not just make the list of testcases

Reply:
> also feels like a make_test_cases or get_test_cases rather than setup_ which I'd expect to setup something for the tests, not just make the list of testcases

True. No objections.